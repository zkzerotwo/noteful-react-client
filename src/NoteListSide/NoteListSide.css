.NoteListSide__back-button {
    margin: 45px 20px 25px auto
}

.NoteListSide__back-button:hover {
    background-color: #FFB72F;
}

.NoteListSide__folder-name {
    color: #b5b3b3;
    font-family: var(--custom-font);
    font-size: 40px;
    height: 50vh;
    margin: 0 0 0 auto;
    text-orientation: upright;
    width: 75px;
    writing-mode: vertical-lr;
}