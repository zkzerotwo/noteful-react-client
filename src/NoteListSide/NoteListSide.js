import React from 'react'
import PropTypes from 'prop-types'
import { findNote, findFolder } from '../folderFinders'
import NotefulContext from '../NotefulContext'
import CircleButton from '../CircleButton/CircleButton'

class NoteListSide extends React.Component {
    static defaultProps = {
        history: {
            goBack: () => { }
        },
        match: {
            params: {}
        }
    }
    static contextType = NotefulContext;

    render() {
        const { notes, folders } = this.context
        const { noteId } = this.props.match.params;
        const note = findNote(notes, noteId) || {};
        const folder = findFolder(folders, note.folderId);
        return (
            <div className='NoteListSide'>
                <CircleButton
                    tag='button'
                    role='link'
                    onClick={() => this.props.history.goBack()}
                    className='NoteListSide__back-button'
                >
                    <br />
        Back
      </CircleButton>
                {folder && (
                    <h3 className='NoteListSide__folder-name'>
                        {folder.name}
                    </h3>
                )}
            </div>
        )
    }
}


export default NoteListSide

NoteListSide.propTypes = {
    notes: PropTypes.array,
    folders: PropTypes.array,
    noteId: PropTypes.string,
}