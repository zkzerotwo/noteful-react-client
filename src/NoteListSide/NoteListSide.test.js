import React from 'react';
import { shallow } from 'enzyme'
import  Enzyme  from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import NoteListSide from './NoteListSide'

Enzyme.configure({ adapter: new Adapter() });
describe(`NoteListSide component`, () => {
    const props = {
        folder: {
            "name": "Important"
        }
    }

    it('renders a .NoteListSide by default', () => {
        const wrapper = shallow(<NoteListSide />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders a h3 with folder name when in props', () => {
        const h3 = shallow(<NoteListSide {...props} />)
            .find('.NoteListSide__folder-name')
        expect(toJson(h3)).toMatchSnapshot()
    })
})