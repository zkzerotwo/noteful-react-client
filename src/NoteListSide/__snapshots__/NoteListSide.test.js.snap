// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteListSide component renders a .NoteListSide by default 1`] = `
<div
  className="NoteListSide"
>
  <NavCircleButton
    className="NoteListSide__back-button"
    onClick={[Function]}
    role="link"
    tag="button"
  >
    <br />
    Back
  </NavCircleButton>
</div>
`;

exports[`NoteListSide component renders a h3 with folder name when in props 1`] = `null`;
